src/Plugin.php

```diff
@@ -100,11 +100,15 @@ function ($code, $message, $file, $line, $context = null) use (&$previousHandler
                         if (PHP_VERSION_ID >= 80000) {
                             $trace = debug_backtrace();
                             foreach ($trace as $idx => $traceEntry) {
-                                if ($traceEntry['function'] !== 'trigger_error') {
+                                if ($traceEntry['function'] !== 'deprecationWarning') {
                                     continue;
                                 }
-                                $file = $trace[$idx + 2]['file'];
-                                $line = $trace[$idx + 2]['line'];
+                                $offset = 1;
+                                if (isset($traceEntry['args'][1])) {
+                                    $offset = $traceEntry['args'][1];
```

Comment:
`1` must be `2` in next version that supports the cake 5

Reply:
Will a test fail so we don't forget?

Reply:
added a test